wlauto/common/android/device.py

```diff
@@ -670,6 +670,19 @@ def get_device_model(self):
         except KeyError:
             return None
 
+    def is_network_connected(self):
+            network_host = 'www.google.com'
+            self.logger.debug('Checking for internet connectivity.')
+            output = adb_shell(self.adb_name, 'ping -q -w 1 -c 1 {}'.format(network_host),
```

Comment:
On 26 Jul 2016, 02:16:30 -0700, johnrchrdsn wrote:
> > @@ -670,6 +670,19 @@ def get_device_model(self):
> >          except KeyError:
> >              return None
> >
> > +    def is_network_connected(self):
> > +            network_host = 'www.google.com'
> > +            self.logger.debug('Checking for internet connectivity.')
> > +            output = adb_shell(self.adb_name, 'ping -q -w 1 -c 1 {}'.format(network_host),
>
> I believe this is present on all android devices and is part of the standard binaries included. It is part of busybox anyway (which I believe is a requirement for WA?) so it should be in the path anyway.

busybox is not a requirement of WA -- it's in WA and will be deployed by
WA itself, however it won't be in PATH. That's where I was headed with
this -- if you're not sure it will be on all devices it would safer to
invoke `ping` via busybox explicitly:

         output = adb_shell(self.adb_name, '{} ping -q -w 1 -c 1 {}'.format(self.busybox, network_host),

>
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/ARM-software/workload-automation/pull/214/files/2c199966e414d33bea8eb8a0c758a716c732039b#r72214248
IMPORTANT NOTICE: The contents of this email and any attachments are confidential and may also be privileged. If you are not the intended recipient, please notify the sender immediately and do not disclose the contents to any other person, use it for any purpose, or store or copy the information in any medium. Thank you.